packages/connection-storage/src/connection-storage.ts

```diff
@@ -98,11 +107,128 @@ export class ConnectionStorage {
     this.calledOnce = true;
   }
 
-  private static mapStoredConnectionToConnectionInfo(
-    storedConnectionInfo: ConnectionInfo,
+  static async migrateToSafeStorage() {
+    // If user lands on this version of Compass with legacy connections (using storage mixin),
+    // we will ignore those and only migrate connections that have connectionInfo.
+    const connections = (await this.getConnections()).filter(
+      (x) => x.connectionInfo
+    );
+    if (connections.length === 0) {
+      log.info(
+        mongoLogId(1_001_000_200),
```

Comment:
```
$ node scripts/check-logids.js 
Next available log id: 1_001_000_270 (1001000270)
```

We’re not re-using old log ids unless they actually refer to the same thing, `1_001_000_200` was previously `Failed to parse connection` (removed in 2c91060412b4).

Reply:
ah! didn't know about this. thanks.

Reply:
fixed in [7d055e0](https://github.com/mongodb-js/compass/pull/5115/commits/7d055e0ceb031894ffd1eb55dea4fe0225dd9f38)